### DISCUSSION TITLE
Not able to import DefaultOptions type on latest version

### DISCUSSION THREAD

**Discussion by PabloFNK** [TanStack/query#3029] [Q&A] [answered]
Hi!

I am using an old version of the library on my project (version `3.16.0`) and I was updating to the latest when I have found something that has stopped working.

I have the next line on my project:

```
import { DefaultOptions } from 'react-query/types/core/types';
```

Updating to the latest version I am getting the next error:

```
Cannot find module 'react-query/types/core/types' or its corresponding type declarations.

    1 import { DefaultOptions } from 'react-query/types/core/types';
```

I have been checking and this import works until the version `3.27.0`. From there I get this error.

The weird thing is that the type file is in there and exporting that interface. Checking the changes between that version an the next one I am not really able to get what has changed to make this fail.

Do you know what is going on?

**Comment by TkDodo** [comment, marked as answer]
just tried this in a codesandbox and I can import `DefaultOptions` either directly from `react-query` (preferred) or from the path you've shown on the latest RQ version:

https://codesandbox.io/s/ecstatic-morning-brgd1?file=/pages/index.tsx

So that would be a "cannot reproduce" from my side 🤷

**Reply by PabloFNK** [reply]
I have found out how to reproduce. We are using typescript version '3.9.9', if you change your sandbox to use that Typescript version it will fail.

Specifically I think the change from that version that is making this fail for that typescript version is this one:
https://github.com/tannerlinsley/react-query/commit/2c1d74baf79ac2faf413eae3263e8c3348aca1a8#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

This is not working:
```
import { DefaultOptions } from 'react-query/types/core/types';
```

This works:
```
import { DefaultOptions } from 'react-query';
```

**Reply by TkDodo** [reply]
then please import them directly from `react-query` :)
